packages/core/src/create.ts

```diff
@@ -378,13 +378,20 @@ export function envelop({ plugins }: { plugins: Plugin[] }): Envelop {
           contextValue: context,
         });
 
-        for (const afterCb of afterCalls) {
-          afterCb({
-            result,
-            setResult: newResult => {
-              result = newResult;
-            },
-          });
+        const isAsyncIterable = (input: unknown): input is AsyncIterableIterator<any> =>
+          typeof input === 'object' && input != null && Symbol.asyncIterator in input;
+
+        if (!isAsyncIterable(result)) {
+          for (const afterCb of afterCalls) {
+            afterCb({
+              result,
+              setResult: newResult => {
+                result = newResult;
+              },
+            });
+          }
+        } else {
+          // TODO: implement onExecuteDone invocation for async iterables
```

Comment:
Had to adjust this, as it is currently not clear how we should handle AsyncIterables returned from `execute`. See `onExecuteDone` section over here: https://github.com/dotansimha/envelop/issues/182

Reply:
I removed this for now.